x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/conversations/append_conversation_messages.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+
+import { ConversationResponse, Message } from '@kbn/elastic-assistant-common';
+import { getConversation } from './get_conversation';
+
+export interface AppendConversationMessagesParams {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  conversationIndex: string;
+  existingConversation: ConversationResponse;
+  messages: Message[];
+}
+
+export const appendConversationMessages = async ({
+  esClient,
+  logger,
+  conversationIndex,
+  existingConversation,
+  messages,
+}: AppendConversationMessagesParams): Promise<ConversationResponse | null> => {
+  const updatedAt = new Date().toISOString();
+
+  const params = transformToUpdateScheme(updatedAt, [
+    ...(existingConversation.messages ?? []),
+    ...messages,
+  ]);
+  try {
+    const response = await esClient.updateByQuery({
+      conflicts: 'proceed',
```

Comment:
I'm wondering if overriding the default behavior of `conflicts`, (which defaults to `abort`, per https://www.elastic.co/guide/en/elasticsearch/reference/8.12/docs-update-by-query.html) is correct here. Should append still update the conversation if a newer version of the document exists?